src/scripts/main.js

```diff
@@ -1,7 +1,224 @@
 'use strict';
 
-// Uncomment the next lines to use your game instance in the browser
-// const Game = require('../modules/Game.class');
-// const game = new Game();
+const SIZE_BOARD = 4;
+let board = [...Array(SIZE_BOARD)].map(() => Array(SIZE_BOARD).fill(0));
+let score = 0;
+let isWin = false;
+const cells = document.querySelectorAll('.field-cell');
+const startButton = document.getElementById('start-button');
+const allowedKeys = ['ArrowLeft', 'ArrowRight', 'ArrowUp', 'ArrowDown'];
+const message = {
+  Start: document.getElementById('message-start'),
+  Lose: document.getElementById('message-lose'),
+  Win: document.getElementById('message-win'),
+};
 
-// Write your code here
+startButton.addEventListener('click', startGame);
+
+function startGame() {
+  resetGame();
+  changeButton();
+  addRandomField(board);
+  addRandomField(board);
+  drawBoard(board, cells);
+}
+
+const resetGame = () => {
+  score = 0;
+  isWin = false;
+  document.querySelector('.game-score').innerText = score;
+  board = [...Array(SIZE_BOARD)].map(() => Array(SIZE_BOARD).fill(0));
+  message.Win.classList.add('hidden');
+  message.Lose.classList.add('hidden');
+};
+
+function addRandomField(boardArray) {
+  const emptyCells = [];
+
+  boardArray.forEach((row, rowIndex) => {
+    row.forEach((value, colIndex) => {
+      // eslint-disable-next-line no-unused-expressions
+      value === 0 && emptyCells.push({ rowIndex, colIndex });
+    });
+  });
+
+  if (emptyCells.length > 0) {
+    const randomIndex = Math.floor(Math.random() * emptyCells.length);
+    const randomCell = emptyCells[randomIndex];
+
+    boardArray[randomCell.rowIndex][randomCell.colIndex] =
+      Math.random() < 0.9 ? 2 : 4;
+  }
+}
+
+const drawBoard = (boardArray, curCells) => {
+  boardArray.forEach((row, rowInd) => {
+    row.forEach((value, colInd) => {
+      const cell = curCells[rowInd * boardArray.length + colInd];
+
+      if (value > 0) {
+        cell.textContent = value;
+        cell.classList = 'field-cell' + ` field-cell--${value}`;
+      } else {
+        cell.textContent = null;
+        cell.classList = 'field-cell';
+      }
+    });
+  });
+};
+
+const changeButton = () => {
+  startButton.classList += ' restart';
+  startButton.innerText = 'Restart';
+  startButton.style = 'border: 2px solid red; color: #776e65; outline: none;';
+  message.Start.classList.add('hidden');
```

Comment:
The 'changeButton' function changes the button text and style to 'Restart'. Ensure that the button's functionality is also updated to restart the game when clicked again.